### PR TITLE
[@astrojs/image] Build all optimized images to dist/assets

### LINKED ISSUE #4677: [@astrojs/image] images should be built to `dist/assets`
### What version of `astro` are you using?

1.0.7

### Are you using an SSR adapter? If so, which one?

none

### What package manager are you using?

pnpm

### What operating system are you using?

mac

### Describe the Bug

Optimized images built for SSG builds are being dropped in the main `dist` directory, they should go into `dist/assets` along side other static assets built by Vite

### Link to Minimal Reproducible Example

https://github.com/withastro/www-2

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by tony-sull**
## Changes

Closes #4677 

Fixes a bug that that was building optimized images in the main `dist` directory during SSG builds.  These should be put in `dist/assets` along side other static assets built by Vite

## Testing

Tests updated to expect the `dist/assets` directory

## Docs

None, bug fix only

### CODE REVIEW

File: packages/integrations/image/src/build/ssg.ts [hunk 1]
```diff
@@ -83,10 +83,10 @@ export async function ssgBuild({ loader, staticImages, config, outDir, logLevel
 			let outputFile: string;
 
 			if (isRemoteImage(src)) {
-				const outputFileURL = new URL(path.join('./', path.basename(filename)), outDir);
+				const outputFileURL = new URL(path.join('./assets', path.basename(filename)), outDir);
```
Comment (natemoo-re) [inline packages/integrations/image/src/build/ssg.ts:86]: Can I have a custom `outDir` like `dist/assets/img/`? Seems like I'd want to control that so I don't always get `dist/assets/img/assets/...`?

**Review by natemoo-re — COMMENTED** [review]
Looks good! Is there any sense in introducing an option to customize this as long as we're looking at this logic?

File: packages/integrations/image/src/build/ssg.ts (hunk 1 above)
Comment (tony-sull) [reply packages/integrations/image/src/build/ssg.ts:86]: Yep I definitely want to add that config, I had considered doing it in this PR but wanted to get the bug fix out the door without blocking on a new feature.

Happy to revisit that and add a config option here if you think it's worth doing both here though 👍

Comment (natemoo-re) [reply packages/integrations/image/src/build/ssg.ts:86]: Fine with a quick bugfix to get out and revisiting this in a follow-up!

**Review by natemoo-re — APPROVED** [review]

**Pull request merged**